src/pages/index.js

```diff
@@ -10,33 +10,75 @@ import PopupWithForm from '../components/PopupWithForm.js';
 import PopupWithImages from '../components/PopupWithImages.js';
 import UserInfo from '../components/UserInfo.js';
 import FormValidator from '../components/FormValidator.js';
+import PopupWithConfirmation from '../components/PopupWithConfirmation.js';
+import Api from '../components/Api.js';
+
+
 
 const profileEditButton = document.querySelector('#profile-edit-button');
 const addNewCardButton = document.querySelector('#profile-add-button');
 const profileNameInput = document.querySelector('#profile-name-input');
 const profileDescriptionInput = document.querySelector('#profile-description-input');
 const editProfileForm = document.forms['profile-form'];
 const addCardForm = document.forms['card-form'];
+const avatarContainer = document.querySelector('.profile__avatar-container');
+
 
 const popupWithImage = new PopupWithImages('#full-image-modal');
 popupWithImage.setEventListeners();
 
 const userInfo = new UserInfo({
   nameSelector: '#profile-name',
-  descriptionSelector: '#profile-description'
+  descriptionSelector: '#profile-description',
+  avatarSelector: '.profile__image'
 });
 
-function createCard({ name, link }) {
+
+function createCard(cardData) {
   const card = new Card(
-    { name, link },
+    cardData,
     '#card-template',
-    (title, imageUrl) => {
-      popupWithImage.open(title, imageUrl);
+    {
+      handleCardClick: (title, imageUrl) => {
+        popupWithImage.open(title, imageUrl);
+      },
+      handleDeleteClick: (cardId, cardElement) => {
+        deleteCardPopup.open(cardId, cardElement);
+      },
+      handleLikeClick: (cardId, isLiked, cardInstance) => {
+        if (isLiked) {
+          api.unlikeCard(cardId)
+            .then(updatedCardData => {
+              cardInstance.updateLikes(updatedCardData.likes);
+            })
+            .catch(err => console.error(err));
+        } else {
+          api.likeCard(cardId)
+            .then(updatedCardData => {
+              cardInstance.updateLikes(updatedCardData.likes);
+            })
+            .catch(err => console.error(err));
+        }
+      }
     }
   );
   return card.getView();
 }
 
+
+const deleteCardPopup = new PopupWithConfirmation('#delete-confirm-modal', {
+  handleFormSubmit: (cardId, cardElement) => {
+    api.deleteCard(cardId)
+      .then(() => {
+        cardElement.remove();
+        deleteCardPopup.close();
+      })
+      .catch(err => console.error(err));
+  }
+});
+deleteCardPopup.setEventListeners();
+
+
 const cardSection = new Section(
   {
     items: initialCards,
```

Comment:
`items: initialCards,`

Please, delete it

You don't need `initialCards` anymore because you should render only the server cards

Reply:
* My remark hasn't been corrected https://github.com/XGLogan/se_project_aroundtheus/pull/1/files#r1986434791


 You don't need `initialCards` anymore because you should render only the server cards . That's why there are errors while liking and deleting

Your code will be like this:

```js


const cardSection = new Section(
  {
    renderer: (cardItem) => {
      const cardElement = createCard(cardItem);
      cardSection.addItem(cardElement);
    },
  },
  ".cards__list"
);
```